File: backend/app/schemas/notification.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: UUID
    user_id: UUID
    type: str
    severity: str
    title: str
    message: str | None
    is_read: bool
    reference_id: UUID | None
    reference_type: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationList(BaseModel):
    items: list[NotificationResponse]
    total: int
    unread_count: int

File: frontend/src/types/notification.ts
export type NotificationType =
  | 'critical_threat'
  | 'quarantine_pending'
  | 'false_positive'
  | 'pipeline_health'
  | 'system'

export type NotificationSeverity = 'info' | 'warning' | 'critical'

export interface Notification {
  id: string
  user_id: string
  type: NotificationType
  severity: NotificationSeverity
  title: string
  message: string | null
  is_read: boolean
  reference_id: string | null
  reference_type: string | null
  created_at: string
}

export interface NotificationList {
  items: Notification[]
  total: number
  unread_count: number
}
